File: src/main/java/se/rsrp/hometeleport/events/PlayerLeave.java
package se.rsrp.hometeleport.events;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

import se.rsrp.hometeleport.TeleportManager;

public class PlayerLeave implements Listener {
	private final TeleportManager teleportManager;

	public PlayerLeave(TeleportManager teleportManager) {
		this.teleportManager = teleportManager;
	}

	@EventHandler
	public void onPlayerLeave(PlayerQuitEvent event) {
		teleportManager.cancelTeleport(event.getPlayer());
	}
}

File: src/main/kotlin/se/rsrp/hometeleport/events/PlayerLeave.kt
package se.rsrp.hometeleport.events

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent
import se.rsrp.hometeleport.TeleportManager

class PlayerLeave(private val teleportManager: TeleportManager) : Listener {
    @EventHandler
    fun onPlayerLeave(event: PlayerQuitEvent) {
        teleportManager.cancelTeleport(event.player)
    }
}
